pymanopt/manifolds/sphere.py

```diff
@@ -103,70 +84,102 @@ def _normalize(self, X):
         return X / self.norm(None, X)
 
 
-class SphereSubspaceIntersection(Sphere):
-    """
-    Manifold of n-dimensional unit 2-norm vectors intersecting the
-    r-dimensional subspace of R^n spanned by the columns of the matrix U. This
-    implementation is based on spheresubspacefactory.m from the Manopt MATLAB
-    package.
-    """
-
-    def __init__(self, n, U=None):
-        super(SphereSubspaceIntersection, self).__init__(n)
-
-        self._n = n
-        if U is None:
-            self._subspace_projector = np.eye(n)
-            self._subspace_dimension = n
-            # The name is defined in the base class.
-        else:
-            if U.shape[0] != n:
-                raise ValueError(
-                    "Number of rows in U does not match dimension of the "
-                    "ambient space.")
-            self._configure_manifold(U)
+class Sphere(_Sphere):
+    """Manifold of shape n1 x n2 x ... x nk tensors with unit 2-norm. The
+    metric is such that the sphere is a Riemannian submanifold of Euclidean
+    space.
 
-        if self.dim == 0:
-            warnings.warn("Manifold only consists of isolated points when "
-                          "subspace is 1-dimensional.")
+    Notes
+    -----
+    The implementation of the Weingarten map is taken from [1]_.
```

Comment:
Typo? (trailing underscore)

Reply:
This is the way footnotes are defined in rst (see [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes)). Our docstrings right now are very inconsistent in general. I'm planning to clean this up after the backend rewrite by adopting the numpy docstring format.

Reply:
Thanks!